pkg/tournament/classic_division.go

```diff
@@ -500,6 +531,57 @@ func (t *ClassicDivision) AddPlayers(persons *entity.TournamentPersons) error {
 	return nil
 }
 
+func (t *ClassicDivision) RemovePlayers(persons *entity.TournamentPersons) error {
+	for person, _ := range persons.Persons {
+		playerIndex, ok := t.PlayerIndexMap[person]
+		if !ok {
+			return fmt.Errorf("player %s does not exist in"+
+				" classic division RemovePlayers", person)
+		}
+		if playerIndex < 0 || playerIndex >= len(t.Players) {
+			return fmt.Errorf("player index %d for player %s is"+
+				" out of range in classic division RemovePlayers", playerIndex, person)
+		}
+	}
+
+	playersRemaining := len(t.Players)
+	for i := 0; i < len(t.PlayersProperties); i++ {
+		if t.PlayersProperties[i].Removed {
+			playersRemaining--
+		}
+	}
+
+	if playersRemaining-len(persons.Persons) <= 0 {
+		return fmt.Errorf("cannot remove players as tournament would be empty")
+	}
+
+	for person, _ := range persons.Persons {
+		t.PlayersProperties[t.PlayerIndexMap[person]].Removed = true
+	}
+
+	for i := 0; i < len(t.Matrix); i++ {
+		resultsArePresent := true
+		if i == t.CurrentRound {
+			presentResults, err := t.ResultsArePresent(i)
+			if err != nil {
+				return err
+			}
+			resultsArePresent = presentResults
+		}
+		if i > t.CurrentRound || (i == t.CurrentRound && !resultsArePresent) {
+			pm := t.RoundControls[i].PairingMethod
+			if i == t.CurrentRound || pair.IsStandingsIndependent(pm) && pm != entity.Manual {
```

Comment:
which of these is intended? (by default && is ()'ed before ||)
```suggestion
			if i == t.CurrentRound || (pair.IsStandingsIndependent(pm) && pm != entity.Manual) {
```
or
```suggestion
			if (i == t.CurrentRound || pair.IsStandingsIndependent(pm)) && pm != entity.Manual {
```

Reply:
Ah, thanks for pointing that out. The first of your suggestions is the intended expression. I prefer this suggestion over relying on boolean precedence since it makes the code more clear and I don't have to think about boolean precedence. Thanks for catching that.